### ISSUE TITLE
"not enough memory: you tried to allocate 0GB" error is being observed on executin train.lua and after running 390/390 tests

### ISSUE THREAD

**Issue by streamhsa** [szagoruyko/cifar.torch#6]
Hi Sergey,

"not enough memory: you tried to allocate 0GB" error is being observed on executin train.lua and after running 390/390 tests. Does incresing the RAM really help? My current RAM size is 8GB.

Train accuracy: 10.15 %  time: 1388.22 s  
==> doing some iterations on train data for batch normalization 
/root/torch/install/bin/lua: $ Torch: not enough memory: you tried to allocate 0GB. Buy new RAM! at /root/torch/pkg/torch/lib/TH/THGeneral.c:210
stack traceback:
    [C]: ?
    [C]: in function 'Tensor'
    /root/torch/install/share/lua/5.1/cltorch/Random.lua:8: in function 'bernoulli'
    /root/torch/install/share/lua/5.1/nn/Dropout.lua:19: in function 'updateOutput'
    /root/torch/install/share/lua/5.1/nn/Sequential.lua:44: in function 'updateOutput'
    /root/torch/install/share/lua/5.1/nn/Sequential.lua:44: in function 'updateOutput'
    /root/torch/install/share/lua/5.1/nn/Sequential.lua:44: in function </root/torch/install/share/lua/5.1/nn/Sequential.lua:41>
    (tail call): ?
    train.lua:131: in function 'test'
    train.lua:191: in main chunk
    [C]: in function 'dofile'
    .../torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
    [C]: ?

\Thanks

**Comment by streamhsa** [comment]
After increasing the RAM, it worked. However, I see a new issue now :(

==> doing some iterations on train data for batch normalization 
==> testing 
Test accuracy:  10  
/root/torch/install/bin/lua: cannot open </usr/bin/gnuplot -persist > /dev/null 2>&1> in mode  w at /root/torch/pkg/torch/lib/TH/THDiskFile.c:585
stack traceback:
    [C]: ?
    [C]: in function 'PipeFile'
    ...root/torch/install/share/lua/5.1/gnuplot/gnuplot.lua:134: in function 'getfigure'
    ...root/torch/install/share/lua/5.1/gnuplot/gnuplot.lua:651: in function 'filefigure'
    ...root/torch/install/share/lua/5.1/gnuplot/gnuplot.lua:658: in function 'epsfigure'
    /root/torch/install/share/lua/5.1/optim/Logger.lua:163: in function 'plot'
    train.lua:152: in function 'test'
    train.lua:191: in main chunk
    [C]: in function 'dofile'
    .../torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
    [C]: ?

**Comment by szagoruyko** [comment]
hi, you are on opencl branch right? it is a little bit after cuda version, I have added an option that will hopefully fix this. Also about RAM, you could try to add some collectgarbage() calls in the main for loop to save some memory.

**Comment by streamhsa** [comment]
Hi Sergey,

I am now getting the below error :(

==> configuring model  
/root/torch/install/bin/lua: /root/torch/install/share/lua/5.1/nn/Copy.lua:11: attempt to index a nil value
stack traceback:
    /root/torch/install/share/lua/5.1/nn/Copy.lua:11: in function '__init'
    /root/torch/install/share/lua/5.1/torch/init.lua:54: in function </root/torch/install/share/lua/5.1/torch/init.lua:50>
    [C]: in function 'Copy'
    train.lua:45: in main chunk
    [C]: in function 'dofile'
    .../torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
    [C]: ?
